test/lint/lint-python-mutable-default-parameters.sh

```diff
@@ -8,7 +8,7 @@
 
 export LC_ALL=C
 EXIT_CODE=0
-OUTPUT=$(git grep -E '^\s*def [a-zA-Z0-9_]+\(.*=\s*(\[|\{)' -- "*.py")
+OUTPUT=$(git grep -E '^\s*def [a-zA-Z0-9_]+\(.*=\s*(\[|\{)' -- "*.py" ':!*/bitcoin_functional/*')
```

Comment:
Are we planning to just nuke the entire `bitcoin_functional` directory? Can we just do that here instead?

Reply:
Hmm, I don't really know. I understand the original motivation. But it's also annoying to maintain and maybe it doesn't really make sense to try and keep supporting running in Bitcoin mode.

Reply:
For the sake of getting this in, let's leave this the way it is, and nuke bitcoin_functional as a followup.